### ISSUE TITLE
shall I use your `DataParallelWithCallback` instead of pytorch `nn.DataParallel`?

### ISSUE THREAD

**Issue by CoinCheung** [vacancy/Synchronized-BatchNorm-PyTorch#19]
Hi, 

I do not quite understand the examples you released. What if `nn.BatchNorm2d` is part of a module ? 
For example, if I use your `SynchronizedBatchNorm2d` to replace the `nn.BatchNorm2d` in the resnet50 model, shall I use `net = DataParalleWithCallBack(net)` instead of `net = nn.DataParallel(net)` ? Or do I just need to use `DataParallelWithCallBack` to wrap your implemented `SynchronizedBatchNorm2d` ?

**Comment by vacancy** [comment]
Hi, thanks for your question. In short: use `net = DataParalleWithCallBack(net)`.
